### PR TITLE
feature: support init inherit old mosn config function

### PR DESCRIPTION
**Pull request by dengqian**
### Issues associated with this PR

Your PR should present related issues you want to solve.

### Solutions
You should show your solutions about the issues in your PR, including the overall solutions, details and the changes. At this time, Chinese is allowed to describe these.

### UT result
Unit Test is needed if the code is changed, your unit test should cover boundary cases, corner cases, and some exceptional cases. And you need to show the UT result.

### Benchmark
If your code involves the processing of every request, you should give the Benchmark Result.

### Code Style
+ Make sure `Goimports` has run
+ Show `Golint` result

### CODE REVIEW

**Review by doujiang24 — APPROVED** [review]
Seems it keep the backward compability, LGTM.

File: pkg/mosn/mosn.go [hunk 1]
```diff
@@ -100,24 +101,19 @@ func (m *Mosn) inheritHandler() error {
...
 	log.StartLogger.Infof("[mosn] [NewMosn] active reconfiguring")
 	// parse MOSNConfig again
 	c := configmanager.Load(configmanager.GetConfigPath())
 	if c.InheritOldMosnconfig {
-		// inherit old mosn config
-		oldMosnConfig, err := server.GetInheritConfig()
+		err = inheritFunc(c)
 		if err != nil {
 			m.Upgrade.ListenSockConn.Close()
 			log.StartLogger.Errorf("[mosn] [NewMosn] GetInheritConfig failed, exit")
```
Comment (doujiang24) [inline pkg/mosn/mosn.go:114]:
```suggestion
			log.StartLogger.Errorf("[mosn] [NewMosn] InheritConfig failed, exiting, err: %v", err)
```

**Review by doujiang24 — APPROVED** [review]

File: pkg/mosn/mosn.go [hunk 2]
```diff
@@ -126,6 +122,32 @@ func (m *Mosn) inheritHandler() error {
 	return nil
 }
 
+// replace your own inherit func with default inherit func
+func InitInheritFunc(f func (c *v2.MOSNConfig) error) {
```
Comment (taoyuanyuan) [inline pkg/mosn/mosn.go:126]: Where is this func called?

Comment (dengqian) [reply pkg/mosn/mosn.go:126]: > Where is this func called?

If we wants to customize  inheritFunc, we can call this function in main.go to replace the default inheritFunc

**Review by taoyuanyuan — APPROVED** [review]
LGTM

**Pull request merged**